clients/client/src/main/java/org/eclipse/sw360/clients/adapter/SW360VulnerabilityClientAdapterAsyncImpl.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Siemens AG, 2021. Part of the SW360 Portal Project.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.clients.adapter;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.sw360.clients.rest.SW360VulnerabilityClient;
+import org.eclipse.sw360.clients.rest.resource.vulnerabilities.SW360ReleaseVulnerabilityRelation;
+import org.eclipse.sw360.clients.rest.resource.vulnerabilities.SW360Vulnerability;
+import org.eclipse.sw360.clients.utils.FutureUtils;
+
+/**
+ * Adapter implementation for the SW360 Vulnerabilities endpoint.
+ */
+class SW360VulnerabilityClientAdapterAsyncImpl implements SW360VulnerabilityClientAdapterAsync {
+    private final SW360VulnerabilityClient vulnerabilityClient;
+
+    public SW360VulnerabilityClientAdapterAsyncImpl(SW360VulnerabilityClient client) {
+        vulnerabilityClient = client;
+    }
+
+    @Override
+    public SW360VulnerabilityClient getVulnerabilityClient() {
+        return vulnerabilityClient;
+    }
+
+    @Override
+    public CompletableFuture<Optional<SW360Vulnerability>> getVulnerabilityByExternalId(String externalId) {
+        return FutureUtils.optionalFuture(getVulnerabilityClient().getVulnerabilityByExternalId(externalId));
+    }
+
+    @Override
+    public CompletableFuture<List<SW360Vulnerability>> getVulnerabilities() {
+        return getVulnerabilityClient().getVulnerabilities();
+    }
+
+    @Override
+    public CompletableFuture<SW360Vulnerability> createVulnerability(SW360Vulnerability vulnerability) {
+        return FutureUtils.wrapInFuture(() -> {
+            if (vulnerability.getId() != null) {
+                throw new IllegalArgumentException("Vulnerability already has the id " + vulnerability.getId());
+            }
+            return SW360VulnerabilityAdapterUtils.validateVulnerability(vulnerability);
+        }, "Cannot create vulnerability with external Id - " + vulnerability.getExternalId())
+                .thenCompose(getVulnerabilityClient()::createVulnerability);
+    }
+
+    @Override
+    public CompletableFuture<SW360Vulnerability> updateVulnerability(SW360Vulnerability vulnerability,
+            String externalId) {
+        return FutureUtils.wrapInFuture(() -> {
+            if (vulnerability.getExternalId() == null || vulnerability.getExternalId().trim().isEmpty()) {
```

Comment:
Please check if we can use the isBlank() method here for externalId check instead of .trim().isEmpty(). same for below line 79.

Reply:
done